docs/chapters/workflow/git-workflow.rst

```diff
@@ -0,0 +1,57 @@
+
+Git workflow
+************
+
+The suggested workflow when working in a somewhat large project with git, given that one does not
+use gerrit or a similar tool, is a rather general pull request based flow. The following flow is
+general enough so that it works in both GitHub, GitLab and Bitbucket, at least.
+
+Assume in the examples below that the URL to the repo is
+``git@github.com:Pelagicore/software-factory.git``.
+
+General setup
+=============
+#. Fork the repo. That fork will end up at ``git@github.com:<username>/software-factory.git``
+#. Clone the original repo and add a remote for the fork to the repo ::
+
+    $ git clone git@github.com:Pelagicore/software-factory.git
+    $ git remote add <username> git@github.com:<username>/software-factory.git
+
+It can be a good idea to first clone the forked repo and then add the upstream repo as a new remote,
+to avoid accidental pushes to the upstream repo. Such pushes can usually also be avoided by
+protecting branches from pushes as well.
```

Comment:
I see the point about not having the canonical repo defaulting to "origin" etc., but maybe the whole guideline above this section should describe a flow where the fork is cloned and the canonical remote added separately as e.g. "upstream" or some other clear name.